homeassistant/components/mqtt/__init__.py

```diff
@@ -546,6 +491,14 @@ async def async_setup(hass: HomeAssistantType, config: ConfigType) -> bool:
     return True
 
 
+def _merge_config(entry, conf):
+    """Merge configuration.yaml config with config entry."""
+    conf = dict(conf)
+    conf.update(entry.data)
```

Comment:
Shouldn't `configuration.yaml` data take preference?

Reply:
There's code and log warnings specifically to inform the user that the config entry takes preference: https://github.com/home-assistant/core/blob/8806470be25a959c58a924eac7e805a727beb20e/homeassistant/components/mqtt/__init__.py#L512-L520